### DISCUSSION TITLE
How to add in multiple Health Int Components?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1038] [Q&A] [answered]
Hi:

Is there any way to add multiple health int or double components using with method?
Currently seems like we could only add in 1 single health int component
and we could add in another health double component but this seems like not a perfect solution
Is there any way we could  add multiple health int components?
If so we could have one for HP and one for MP and others
like this 
```java
entity
  .with(new HealthIntComponent(), "HP")
  .with(new HealthIntComponent(), "MP");
```

Thanks

**Comment by chengenzhao** [comment, marked as answer]
hmmm...
I think I found the answer
just write a class to extend RechargeableDoubleComponent or RechargeableIntComponent then it will work

**Reply by AlmasB** [reply]
Yes you are correct.
